ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java

```diff
@@ -1133,6 +1162,41 @@ private List<DatasetVersionRecord> executeListAllValidVersionStatement(PreparedS
     }
   }
 
+  /**
+   * Execute listValidDatasetsSql statement.
+   * @param statement the listValidDatasetsSql statement.
+   * @param accountId the id for the parent account.
+   * @param containerId the id of the container.
+   * @param pageToken the page token.
+   * @return the page of the dataset names.
+   * @throws SQLException
+   */
+  private Page<String> executeListAllValidDatasetsStatement(PreparedStatement statement, int accountId,
+      int containerId, String pageToken) throws SQLException {
+    ResultSet resultSet = null;
+    try {
+      statement.setInt(1, accountId);
+      statement.setInt(2, containerId);
+      statement.setString(3, pageToken);
+      statement.setString(4, pageToken);
+      statement.setInt(5, mySqlAccountServiceConfig.listDatasetsMaxResult + 1);
```

Comment:
I guess the way to implement the paging is that,
We get one entry more than needed. And then we use the last entry as the starting point of the next page. Very nice approach.

Another approach is that we get the exact entries as we needed. And then for the next query we use > instead of >=.

Comparing with these two, I believe we have a reason to choice the first one. May I know the reason?

Reply:
As I checked, typically we will use the next_page_token to fetch the next page, also followed by named blob as well.